.pre-commit-config.yaml

```
@@ -23,8 +23,14 @@ repos:
     - id: trailing-whitespace
       args: [--markdown-linebreak-ext=md]
 
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v3.15.0
+    hooks:
+    - id: pyupgrade
+      args: ["--py39-plus"]
+
   - repo: https://github.com/charliermarsh/ruff-pre-commit
-    rev: 'v0.0.291'
+    rev: 'v0.0.292'
     hooks:
     - id: ruff
       args: [--fix]
```

I think pre-commit auto-exits with non-zero if a file is changed, this just makes it explicit I suppose?
```suggestion
      args: [--fix, --exit-non-zero-on-fix]
```

---

Good catch!